### PR TITLE
[16.0][MIG] account_analytic_required

### PR DESCRIPTION
**Pull request by nguyenminhchien**

### CODE REVIEW

**Comment by yostashiro** [comment]
@nguyenminhchien Thanks for the PR. Can you please check if this module is really needed in 16.0. I thought the applicability settings of the analytic plan should replace the function of this module.

**Comment by nguyenminhchien** [comment]
> @nguyenminhchien Thanks for the PR. Can you please check if this module is really needed in 16.0. I thought the applicability settings of the analytic plan should replace the function of this module.

Thanks for your feedback, i think we should keep this module. The applicability of the analytic plan is used for validating the analytic distribution. while this module adds a constraint to check if the Analytic is set or not, base on the analytic policy .

**Review by StephaneMangin — APPROVED** [review]

**Comment by simahawk** [comment]
> > @nguyenminhchien Thanks for the PR. Can you please check if this module is really needed in 16.0. I thought the applicability settings of the analytic plan should replace the function of this module.
> 
> Thanks for your feedback, i think we should keep this module. The applicability of the analytic plan is used for validating the analytic distribution. while this module adds a constraint to check if the Analytic is set or not, base on the analytic policy .

@alexis-via or @grindtildeath WDYT?

File: account_analytic_required/models/account.py
Comment (yankinmax) [inline account_analytic_required/models/account.py:40]: should we really remove `with_company`?

File: account_analytic_required/models/account.py
Comment (yankinmax) [inline account_analytic_required/models/account.py]: you don't change `self`, why do we need this change?

File: account_analytic_required/models/account.py
Comment (yankinmax) [inline account_analytic_required/models/account.py]: maybe better `analytic_acc_ids` and `analytic_accs` to shorten the vars names

File: account_analytic_required/models/account.py
Comment (nguyenminhchien) [reply account_analytic_required/models/account.py]: i just want to make it explicitly, anyway, i have updated to make no change here

File: account_analytic_required/models/account.py
Comment (nguyenminhchien) [reply account_analytic_required/models/account.py]: updated !

**Review by yankinmax — APPROVED** [review]

**Comment by grindtildeath** [comment]
I did some testing on v16.0 (as I hadn't check the new feature before).
I tend to share @yostashiro 's feeling, with the exception that we don't have an option to _always_ require the analytic selection and the mandatory applies for posted moves (but IMO there isn't much sense to strictly require a definition while the record is still in draft).
The issue I can see with the migrated module is that such a configuration can then either be done through the account account model or the analytic plan.
I may have missed something and there might still be a good reason to migrate as is, but I don't see it yet and would appreciate an explanation with proper use case describing what we cannot do with the standard.

File: account_analytic_required/README.rst
Comment (rafaelbn) [inline account_analytic_required/README.rst:38]: - Go to Accounting -> Configuration and activate: Analytic Accounting
- Go to... (I don't know hoy to continue) 😢

**Review by rafaelbn — CHANGES_REQUESTED** [review]
Please could you improve readme?

I'm unable to find account type to configure this module 😄 

Hoy did you test it? I'm trying it but no success

Thank you!

File: account_analytic_required/README.rst
Comment (nguyenminhchien) [reply account_analytic_required/README.rst:38]: I don't know why you don't see these lines:
```
Example:

If you want to have an analytic account on all your *expenses*,
set the policy to *always* for the account of type *expense*.
If you try to save a journal items with an account of type *expense*
without analytic account, you will get an error message.
```
anw, because "Account type" has been removed, the configuration will be done on the Account via menu: **Invoicing > Configuration > Accounting > Chart of Accounts**.

**Comment by rafaelbn** [comment]
We should review deeper, going account by account (hundreds) activating this is not usable.

**Comment by rafaelbn** [comment]
- [x] @nguyenminhchien we should adapt the required to the new account type rework in https://github.com/odoo/odoo/commit/26b2472f4977ccedbb0b5ed5f08be2c04313fd21

**Comment by rafaelbn** [comment]
Hello @nguyenminhchien , I'm testing again it not easy to configure. In Spain we have by default 132 Expense accounts and 121 Income accounts, repeat 😄 **by default (legal ones)**

**Review by rafaelbn — COMMENTED** [review]
Dear all! I guess this module is not needed any more so migration is not needed.

Rationale

- [x] Analytic account has mandatory the Analytic plan
- [x] You can set Analytic plan to be mandatory, so this is the same of this module.

Video: https://www.loom.com/share/2c6928e3380a4ae8a1802a277065120d

Other opinions from you and @OCA/accounting-maintainers are welcome

In my side should be closed and use standard

**Comment by sbidoul** [comment]
> You can set Analytic plan to be mandatory, so this is the same of this module

@rafaelbn are you sure this constraint is applied in all circumstances? I think @alexis-via mentioned there are cases where the validation is not done by Odoo.

**Comment by rafaelbn** [comment]
> @rafaelbn are you sure this constraint is applied in all circumstances? I think @alexis-via mentioned there are cases where the validation is not done by Odoo.

- If you set analytic plan as mandatory it happens.
- If you set as not mandatory then no

Where @alexis-via said that? Twitter? Is there a link?

In all circumstances?

- Do you mean in maybe other views like assets, and so on?

The module doesn't hurt. I mean, if migrated and installed works with Plans

Know that you comment, maybe using some other OCA modules doesn't work as Odoo don't think about other modules.

I'm not blocking at all! 😄 ❤️ just my test to decrease modules when it is possible.

Best regards!

**Review by rafaelbn — APPROVED** [review]
Functionally tested module works 👍 

I think functionality is in core with planes but maybe I missed something.

Up to you! 😄

**Comment by sbidoul** [comment]
@rafaelbn I was referring to [this tweet](https://twitter.com/alexisdlattre/status/1633606705304018946). I've not investigated myself, though so I don't know the details.

File: account_analytic_required/tests/test_account_analytic_required.py
Comment (leemannd) [inline account_analytic_required/tests/test_account_analytic_required.py]: Any reason to drop this test? Isn't it still in the scope of the module?

Comment (nguyenminhchien) [reply account_analytic_required/tests/test_account_analytic_required.py]: Analytic Distribution is Analytic, no need to make this test anymore.

**Review by StephaneMangin — APPROVED** [review]

File: account_analytic_required/models/account.py
Comment (alexis-via) [inline account_analytic_required/models/account.py:21]: ARF !!! Not on account.account model which is always attached to a specific company !!!

**Pull request merged**
